streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -122,7 +126,15 @@ public static FileLock lockStateDirectory(File stateDir) throws IOException {
 
     private static FileLock lockStateDirectory(File stateDir, int retry) throws IOException {
         File lockFile = new File(stateDir, ProcessorStateManager.LOCK_FILE_NAME);
-        FileChannel channel = new RandomAccessFile(lockFile, "rw").getChannel();
+        FileChannel channel = null;
+        synchronized (channels) {
+            channel = channels.get(lockFile);
+            if (channel == null) {
+                channel = new RandomAccessFile(lockFile, "rw").getChannel();
+                channels.put(lockFile, channel);
```

Comment:
I don't know whether two threads will be competing for the same file lock since each thread's partition list is different.

But if we want to deal with that situation, we can:
1. extend the synchrnozed block to cover lockStateDirectory as well
2. or don't remove the channel object from the map since there are not many those channel objects in the system, about a dozen of them in a usual deployment.